hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/ReadableFileStoreImpl.java

```diff
@@ -57,10 +56,10 @@ public ReadableFileStoreImpl(@NonNull final ReadableStates states) {
      */
     public @Nullable FileMetadata getFileMetadata(@Nullable final FileID id) {
         final var file = getFileLeaf(id);
-        return file.map(FileStore::fileMetaFrom).orElse(null);
+        return file == null ? null : FileStore.fileMetaFrom(file);
     }
 
-    public @Nullable Optional<File> getFileLeaf(@Nullable FileID id) {
-        return Optional.ofNullable(fileState.get(id));
+    public @Nullable File getFileLeaf(@Nullable FileID id) {
```

Comment:
Nit: Not part of this PR, but does it make sense to allow a `@Nullable` `id`?

Reply:
We will do it. That's what I discuss also with @jsync-swirlds

Reply:
and I think you mean @NonNull

Reply:
Already noted for a follow-up.

Reply:
https://github.com/hashgraph/hedera-services/issues/8198 issue created